### ISSUE TITLE
NIOSSHPrivateKey from file/bytes

### ISSUE THREAD

**Issue by WilliamABradley** [apple/swift-nio-ssh#94]
I don't know if I am missing something, is there a way to parse a private key file or private key data with a passphrase into a NIOSSHPrivateKey without knowing the type of private key you want to open?

**Comment by Lukasa** [comment]
Yes, this is currently correct. The main reason we haven't added this support at the moment is that CryptoKit does not provide the algorithms required for us to support protected private keys (that is, private keys protected by passphrases). This is the overwhelmingly most common format for private keys on disk.

I think the most useful addition to work around this would be to add support for the ssh agent protocol, and use that as a potential backing for NIOSSHPrivateKey. This is admittedly a lot of work, but it circumvents the need for us to manage the protection of the key at all, and encourages good SSH key hygiene.

**Comment by WilliamABradley** [comment]
Not if you plan on using this on IOS 😄
